gridGeneration.py

```diff
@@ -0,0 +1,259 @@
+        os.makedirs(os.path.join(folder_path, geojson_path))
+    
+    max_count = 0
+    flag_val = False
+    input_data = None
+    optimal_count_list = None
+    optimal_grid_size_list = None
+    first_depth = 0
+    big_initial_area = 0.0
+
+    if args.maxCount:
+        max_count = int(args.maxCount)
+
+    # read L-2 dissimilarity from stand input
+    for line in sys.stdin.readlines():
+        input_data = ast.literal_eval(line)
+    
+    # save the 2d list as a file
+    with open(os.path.join(folder_path, os.path.basename(input_data['inFolder']) + '.csv' )  , "w") as out_f:
+        writer = csv.writer(out_f)
+        writer.writerows(input_data['nameNum'])
+    
+    # perform the 1st k-d tree
+    for depth_count in range(1, int(maximum_level) + 1):
+        # build k-d tree
+        tree_cons = kdTree(depth_count, input_data['iniBb'], input_data['data'], 1)
+        out_tree = tree_cons.tree_building()
+
+        # get leaves given a K-D tree
+        bb_collec = tree_cons.get_leaves(out_tree)
+
+        # get counts
+        hist, gridid_collec= tree_cons.counts_calculation()
+        # Cell size computation
+        cell_size_computation(depth_count, bb_collec)
+        del tree_cons
+
+        util = Utility(hist)
+        filename = os.path.join(os.path.join(folder_path, path), 'level-' + str(depth_count) + '.png')
+        # probability distribution
+        out_distribution, count_list, count_zero_list, cell_num = util.distribution_computation(filename)
+        # write out a Geojson file
+        util.geojson_write(depth_count, bb_collec, os.path.join(folder_path, geojson_path), cell_num, input_data['iniArea'], None, 'tree_v1', flag_val)
+        del util
+
+        # stop condition (the over 90% (parameter) of cells is less than 10 (parameter) (the count value))
+        if len(count_zero_list) != 0:
+            count_list.insert(0, count_zero_list[0])
+        smallest_max_count = 0
+        smallest_max_count_ind = -1
+        for ind, ele in enumerate(count_list):
+            if ele > count_num:
+                break
+            else:
+                smallest_max_count = ele
+                smallest_max_count_ind = ind
+
+        if smallest_max_count_ind != -1:
+            total_count_within_count_num = 0
+            total_grids = 0
+            list_length = 0
+            if not count_zero_list:  # the list is empty
+                list_length = smallest_max_count_ind + 1
+                total_grids = cell_num
+            else:
+                list_length = smallest_max_count_ind + 2
+                total_grids = cell_num + count_zero_list[1]
+
+            for i in range(list_length):
+                if count_list[i] == 0:
+                    total_count_within_count_num += count_zero_list[1]
+                else:
+                    total_count_within_count_num += out_distribution[count_list[i]]
+
+            if (float(total_count_within_count_num) / float(total_grids)) > grid_percent:
```

Comment:
To retain the decimal places, you actually don't need to cast both of these to float, just one.

Alternatively, you could leverage the `__future__` module which contains division.

Reply:
remove one of the floats.